migration-engine/connectors/sql-migration-connector/src/sql_destructive_change_checker/destructive_change_checker_flavour/mysql.rs

```diff
@@ -49,15 +52,27 @@ impl DestructiveChangeCheckerFlavour for MysqlFlavour {
             return;
         }
 
+        let native_types_enabled = self.features().contains(MigrationFeature::NativeTypes);
+        let datamodel_connector = SqlDatamodelConnectors::mysql();
+        let previous_type = match &columns.previous().column_type().native_type {
+            Some(tpe) if native_types_enabled => datamodel_connector.render_native_type(tpe.clone()),
+            _ => format!("{:?}", columns.previous().column_type_family()),
+        };
+
+        let next_type = match &columns.next().column_type().native_type {
+            Some(tpe) if native_types_enabled => datamodel_connector.render_native_type(tpe.clone()),
+            _ => format!("{:?}", columns.next().column_type_family()),
+        };
+
```

Comment:
note for later: probably some duplication that could be avoided between mysql and postgres